lang_qc/models/pacbio/experiment.py

```diff
@@ -19,14 +19,96 @@
 # You should have received a copy of the GNU General Public License along with
 # this program. If not, see <http://www.gnu.org/licenses/>.
 
-from typing import List
+from typing import Any
 
-from pydantic import BaseModel, ConfigDict, Field
+from pydantic import Field, model_validator
+from pydantic.dataclasses import dataclass
 
 from lang_qc.db.mlwh_schema import PacBioRun
 
 
-class PacBioExperiment(BaseModel):
+@dataclass(kw_only=True, frozen=True)
+class PacBioLibrary:
+    """
+    This model represents LIMS data associated with a PacBio library.
+
+    The fields of the model can be assigned directly via the constructor.
+    However, if the `db_library` field, a single row of the PacBioRun table
+    class, is set via the constructor, the rest of the fields are populated
+    using this database row object, while  any other information passed to the
+    constructor is disregarded.
+
+    The  `db_library` field is not present in the model instance that is
+    returned by the constructor.
+    """
+
+    db_library: PacBioRun = Field(init_var=True)
```

Comment:
Spot check - a "library" is the prep of one sample, therefore 1:1 with a row in `pac_bio_run`? As distinct from pool, which is a pool of libraries?

Reply:
Correct. A pool might contain one library as long as the library is indexed. The term `pool` is only used with indexed libraries.